lib/lambda_wrap/api_gateway_manager.rb

```diff
@@ -151,7 +153,50 @@ def delete_stage(api_id, env)
         puts 'API Gateway stage ' + env + ' does not exist. Nothing to delete.'
       end
     end
-
+
+    # Generate or Update the API Gateway by using the swagger file.
+    # *Arguments*
+    # [api_name] API Gateway name
+    # [local_swagger_file] Path of the local swagger file
+    # [env] Environment
+    def setup_apigateway_by_swagger_file(api_name, local_swagger_file, env)
+
+      #If API gateway with the name is alaredy present then update it else create a new one
+      apis = @client.get_rest_apis(limit: 500).data
+      api = apis.items.select { |a| a.name == api_name }.first
+      swagger_file_content =  File.read(local_swagger_file)
+
+      gateway_response = nil
+      if (api.nil?)
+        gateway_response =  @client.import_rest_api(fail_on_warnings: false, body: swagger_file_content)
+      else
+        gateway_response =  @client.put_rest_api(rest_api_id: api.id, mode: 'merge', fail_on_warnings: false, body: swagger_file_content)
+      end
+
+      if (gateway_response.nil? && gateway_response.id.nil?)
+        raise "Failed to create API gateway with name #{api_name}"
+      else
+        if (api.nil?)
+          puts "Created api gateway #{api_name} having id #{gateway_response.id}"
+        else
+          puts "Updated api gateway #{api_name} having id #{gateway_response.id}"
+        end
+      end
+
+      #Deploy the service
+      stage_variables = {}
+      deployment_description = 'Deployment of service to ' + env
+      stage_variables.store('environment', env)
+      create_stages(gateway_response.id, env, stage_variables)
+
+      service_uri = "https://#{gateway_response.id}.execute-api.#{ENV['AWS_REGION']}.amazonaws.com/#{env}/"

```

Comment:
Can you have a method parameter for the environment, just like in `setup_apigateway`? Same for `stage_variables`.